backends/xnnpack/runtime/XNNPACKBackend.cpp

```diff
@@ -38,12 +57,14 @@ class XnnpackBackend final : public PyTorchBackendInterface {
     // new and since this type is not trivially destructible, we must call the
     // destructor manually in destroy().
     new (executor) xnnpack::delegate::XNNExecutor;
-
+  
+    ET_CHECK_OR_RETURN_ERROR(workspace_ != nullptr, Internal, "Failed to create XNN workspace");
```

Comment:
Should we still fallback to the old behavior of each delegate instance having its own workspace, if the global workspace is nullptr?

Reply:
I left a comment about BC in the summary. V4 has been in XNNPACK for two years now, so should be OK to use it. And this transparent to end users i.e. older PTE will just run faster.

Reply:
This check is mainly because we don't want to raise from the constructor. This isn't a realistic issue, and if it happen old version would also fail similarly just a bit deeper inside XNNPACK.